builders/container-apps-internal-registry-demo/startup-script.sh

```diff
@@ -52,49 +52,124 @@ token=$(printf "%s" "$REGISTRY_AUTH_USERNAME:$REGISTRY_AUTH_PASSWORD" | base64)
 acr_access_string="Basic $token"
 export CNB_REGISTRY_AUTH='{"'$ACR_RESOURCE_NAME'":"'$acr_access_string'"}'
 
+echo "Initiating buildpack build..."
+echo "Correlation id: '$CORRELATION_ID'"
+echo 
+
+RETRY_DELAY=2
+RETRY_ATTEMPTS=5
+
+function fail_if_retry_exceeded() {
+  retries=$1
+  if [ "$retries" -ge $RETRY_ATTEMPTS ]; then
+    echo "----- Retry attempts exceeded -----"
+    exit 1
+  fi
+}
+
 # Execute the analyze phase
 echo
-echo "======================================="
 echo "===== Executing the analyze phase ====="
-echo "======================================="
-/lifecycle/analyzer \
-  -log-level debug \
-  -run-image mcr.microsoft.com/oryx/builder:stack-run-debian-bullseye-20230926.1 \
-  $APP_IMAGE
+retryCount=0
+until [ "$retryCount" -ge $RETRY_ATTEMPTS ]
+do
+  if [ "$retryCount" -ge 1 ]; then
+    echo "----- Retrying analyze phase (attempt $retryCount) -----"
+  fi
+
+  /lifecycle/analyzer \
+    -log-level debug \
+    -run-image mcr.microsoft.com/oryx/builder:stack-run-debian-bullseye-20230926.1 \
+    $APP_IMAGE \
+    && break
+
+  retryCount=$((retryCount+1))
```

Comment:
Nit: not a blocker, just curious: do we need the double parentheses?

Reply:
https://tldp.org/LDP/abs/html/arithexp.html